handler/interface.go

```diff
@@ -17,6 +17,7 @@ type FileManager interface {
 
 type DatasetAPIClient interface {
 	GetVersion(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetID, edition, version string) (dataset.Version, error)
+	GetVersionMetadataSelection(ctx context.Context, req dataset.GetVersionMetadataSelectionInput) (*dataset.Metadata, error)
```

Comment:
We don't need to return a pointer... Inefficiency at memory management as this needs to be placed in the heap to be tracked by the GC.
Not your change...